account/manager.go

```diff
@@ -1754,105 +1818,112 @@ func valueAfterAccountUpdate(account *Account, outputs []*wire.TxOut,
 // the inputs is also provided to use when coming across an unexpected failure.
 // If needed, a change output from the backing lnd node's wallet may be returned
 // as well.
-func (m *manager) inputsForDeposit(ctx context.Context,
-	depositAmount btcutil.Amount, witnessType witnessType,
-	feeRate chainfee.SatPerKWeight) ([]chanfunding.Coin, func(),
-	*wire.TxOut, error) {
+func (m *manager) inputsForDeposit(ctx context.Context, account *Account,
+	newAccountOutput *wire.TxOut, depositAmount btcutil.Amount,
+	witnessType witnessType, feeRate chainfee.SatPerKWeight) (*psbt.Packet,
+	func(), error) {
+
+	// Unfortunately the FundPsbt call doesn't allow us to specify _any_
+	// inputs, otherwise it won't perform coin selection at all. So what we
+	// do instead is to fund our account output just for the funding amount
+	// plus whatever we need to pay for the additional input (which we know
+	// exactly how big it will be). Then we add the account input and its
+	// value to the account output.
+	var acctInputEstimator input.TxWeightEstimator
+	err := addBaseAccountInputWeight(&acctInputEstimator, witnessType)
+	if err != nil {
+		return nil, nil, err
+	}
+	acctInputWeight := int64(acctInputEstimator.Weight())
+	acctInputFee := feeRate.FeeForWeight(acctInputWeight)
 
-	// We'll start by obtaining our global lock ID.
-	lockID, err := m.cfg.Store.LockID()
+	outputToFund := &wire.TxOut{
+		Value:    int64(depositAmount + acctInputFee),
```

Comment:
This _should_ be ok, but part of me wonders if there's some weird edge case here w/ rounding and amounts. Worst case, I guess we get rejected due to having insufficient fees.

Reply:
Yeah, we _should_ always over estimate by a little bit anyway since most of the time the signatures don't end up being the full 73 bytes. But I went ahead and added a min relay fee check anyway.